File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {

        int end = nums1.length-1;

        while(n-1>=0){
            if (m-1<0){
                nums1[end]=nums2[n-1];
                n-=1;
            }
            else if(nums1[m-1]>=nums2[n-1]){
                nums1[end]=nums1[m-1];
                m-=1;
            }
            else{
                nums1[end]=nums2[n-1];
                n-=1;
            }
            end-=1;
        }

        
    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """

        i = m - 1  # real end index of num1
        j = n - 1  # real end indec of num2
        k = m + n - 1  # end of num1

        while j >= 0:
            if i < 0:
                nums1[k] = nums2[j]
                j-=1
            elif nums1[i] >= nums2[j]:
                nums1[k] = nums1[i]
                i-=1
            else:
                nums1[k] = nums2[j]
                j -= 1
            k -= 1

        return nums1
